### PR TITLE
{2023.06}[foss/2021a] Arrow V6.0.0

### PR DESCRIPTION
**Pull request by TopRichard**
edit (by @boegel - 2023-09-06): missing installations for `Arrow-6.0.0-foss-2021a.eb` in EESSI pilot 2023.06:

```
6 out of 53 required modules missing:

* RapidJSON/1.1.0-GCCcore-10.3.0 (RapidJSON-1.1.0-GCCcore-10.3.0.eb)
* RE2/2022-02-01-GCCcore-10.3.0 (RE2-2022-02-01-GCCcore-10.3.0.eb)
* utf8proc/2.6.1-GCCcore-10.3.0 (utf8proc-2.6.1-GCCcore-10.3.0.eb)
* ICU/69.1-GCCcore-10.3.0 (ICU-69.1-GCCcore-10.3.0.eb)
* Boost/1.76.0-GCC-10.3.0 (Boost-1.76.0-GCC-10.3.0.eb)
* Arrow/6.0.0-foss-2021a (Arrow-6.0.0-foss-2021a.eb)
```

### CODE REVIEW

**Comment by TopRichard** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by TopRichard** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by boegel** [comment]
Previous attempt failed to build `pyarrow` with:

```
error: passing const std::vector<std::__cxx11::basic_string<char> > as this argument discards qualifiers [-fpermissive]
  164697 |   __pyx_cur_scope->__pyx_genexpr_arg_0 = __pyx_genexpr_arg_0;
         |                                          ^~~~~~~~~~~~~~~~~~~
  In file included from /cvmfs/pilot.eessi-hpc.org/versions/2023.06/software/linux/x86_64/generic/software/GCCcore/10.3.0/include/c++/10.3.0/vector:72,
                   from /tmp/eb-pvqsw2q9/eb-hvxwnqby/eb-q5cmbnqf/eb-s5tuxpc2/eb-__sjce5p/eb-qqeoear4/eb-33922t8v/eb-2xp5zz8z/eb-u1dcfqsb/eb-ty7vyg0z/pip-req-build-znsnpia7/build/temp.linux-x86_64-cpython-39/lib.cpp:1132:
  /cvmfs/pilot.eessi-hpc.org/versions/2023.06/software/linux/x86_64/generic/software/GCCcore/10.3.0/include/c++/10.3.0/bits/vector.tcc:198:5: note:   in call to std::vector<_Tp, _Alloc>& std::vector<_Tp, _Alloc>::operator=(const std::vector<_Tp, _Alloc>&) [with _Tp = std::__cxx11::basic_string<char>; _Alloc = std::allocator<std::__cxx11::basic_string<char> >]
    198 |     vector<_Tp, _Alloc>::
        |     ^~~~~~~~~~~~~~~~~~~
  gmake[2]: *** [CMakeFiles/lib.dir/build.make:76: CMakeFiles/lib.dir/lib.cpp.o] Error 1
  gmake[1]: *** [CMakeFiles/Makefile2:446: CMakeFiles/lib.dir/all] Error 2
  gmake: *** [Makefile:91: all] Error 2
  error: command '/cvmfs/pilot.eessi-hpc.org/versions/2023.06/software/linux/x86_64/generic/software/CMake/3.20.1-GCCcore-10.3.0/bin/cmake' failed with exit code 2
  ----------------------------------------
  ERROR: Failed building wheel for pyarrow
Failed to build pyarrow
```

It looks like this could be fixed by https://github.com/easybuilders/easybuild-easyconfigs/pull/18348, so we need a `--from-pr` here?

**Comment by TopRichard** [comment]
I don't have access to the logs, so as a first step, i tried rebuilding it, as it might be a failing to download or ...

**Comment by TopRichard** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by TopRichard** [comment]
The package is successfully built, yet the bot shows a failure.

**Comment by boegel** [comment]
@TopRichard The bot reporting a failure is because it detected missing `git` installations, those were fixed in #326.

We now also manually removed the `ReFrame` installations so we can rebuild them in #311.

Once the latter is deployed & merged, we can resume here by rebuilding for all CPU targets, should work as expected...

**Comment by boegel** [comment]
bot: build repo:eessi-2023.06-software

**Comment by boegel** [comment]
Failed build for `RapidJSON-1.1.0-GCCcore-10.3.0.eb` on `aarch64/generic` due to:
```
cc1plus: error: switch -mcpu=generic conflicts with -march=armv8.2-a+crypto+fp16+rcpc+dotprod+ssbs switch [-Werror]
```

Similar issue with the `x86_64/generic` build actually, although the build there doesn't fail, but the build was done using `-march=x86-64 -mtune=generic -fno-math-errno -march=native` (and last option wins), so it's not actually a generic build at all.

Should be fixed with https://github.com/easybuilders/easybuild-easyconfigs/pull/18725

File: eessi-2023.06-eb-4.7.2-2021a.yml [hunk 1]
```diff
@@ -13,3 +13,6 @@ easyconfigs:
        # see https://github.com/easybuilders/easybuild-easyconfigs/pull/18087
        options:                                  
          from-pr: 18087
+  - Arrow-6.0.0-foss-2021a.eb:
+       options:
+         from-pr: 18348
```
Comment (boegel) [inline eessi-2023.06-eb-4.7.2-2021a.yml:18]:
```suggestion
  - RapidJSON-1.1.0-GCCcore-10.3.0.eb:
       # strip out hardcoded -march=native,
       # see https://github.com/easybuilders/easybuild-easyconfigs/pull/18725
       options:
         from-pr: 18725
  - Arrow-6.0.0-foss-2021a.eb:
       # fix installation of pyarrow Python bindings
       # see https://github.com/easybuilders/easybuild-easyconfigs/pull/18348
       options:
         from-pr: 18348

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
bot: build repo:eessi-2023.06-software

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
all staging PRs merged so deployment of this is happening very soon now

**Pull request merged**